openrr-gui/src/joint_position_sender.rs

```diff
@@ -641,3 +641,296 @@ fn round_f64(n: f64) -> f64 {
+        assert_eq!(
+            joint["dummy_joint2"].parent.link,
+            String::from("dummy_parent2")
+        );
+    }
+
+    #[test]
+    fn test_validate_joints() {
+        let joints = dummy_joint();
+
+        let mut raw_joint_trajectory_clients = HashMap::from([(
+            String::from("dummy"),
+            Arc::new(DummyJointTrajectoryClient::new(vec![String::from(
+                "dummy_joint1",
+            )])) as Arc<dyn JointTrajectoryClient>,
+        )]);
+        let speakers = HashMap::from([(
+            String::from("speaker"),
+            Arc::new(DummySpeaker::new()) as Arc<dyn Speaker>,
+        )]);
+
+        let ok_client = dummy_robot_client(raw_joint_trajectory_clients.clone(), speakers.clone());
+
+        raw_joint_trajectory_clients
+            .entry(String::from("err"))
+            .or_insert(Arc::new(DummyJointTrajectoryClient::new(vec![String::from(
+                "no_exist_joint",
+            )])) as Arc<dyn JointTrajectoryClient>);
+
+        let err_client = dummy_robot_client(raw_joint_trajectory_clients, speakers);
+
+        assert!(validate_joints(&joints, &ok_client).is_ok());
+        assert!(validate_joints(&joints, &err_client).is_err());
+    }
+
+    #[test]
+    fn test_error() {
+        let mut errors = Errors {
+            joint_states: Some((10, String::from("ten"))),
+            duration_input: Some(String::from("duration")),
+            other: Some(String::from("option")),
+            update_on_error: false,
+        };
+
+        assert!(!errors.skip_update(&Message::DurationTextInputChanged(String::new())));
+        assert!(!errors.skip_update(&Message::SliderChanged {
+            index: 10,
+            position: 2.
+        }));
+        assert!(!errors.skip_update(&Message::ZeroButtonPressed));
+        assert!(errors.skip_update(&Message::PickListChanged(String::new())));
+        assert!(errors.update_on_error);
+
+        errors.joint_states = None;
+        errors.duration_input = None;
+        assert!(!errors.skip_update(&Message::ZeroButtonPressed));
+        assert!(errors.is_none());
+    }
+
+    #[test]
+    fn test_joint_position_sender() {
+        let manifest_dir = Path::new(env!("CARGO_MANIFEST_DIR"));
+        let theme_path = manifest_dir.join("dark_theme.toml");
+
+        let raw_joint_trajectory_clients = HashMap::from([(
+            String::from("dummy"),
+            Arc::new(DummyJointTrajectoryClient::new(vec![String::from(
+                "dummy_joint1",
+            )])) as Arc<dyn JointTrajectoryClient>,
+        )]);
+        let speakers = HashMap::from([(
+            String::from("speaker"),
+            Arc::new(DummySpeaker::new()) as Arc<dyn Speaker>,
+        )]);
+
+        let robot_client = dummy_robot_client(raw_joint_trajectory_clients, speakers);
+
+        let joint_position_sender = JointPositionSender::new(
+            robot_client,
+            dummy_joint(),
+            Theme::from_file(theme_path).unwrap(),
```

Comment:
I think you can use [style::theme](https://github.com/openrr/openrr/blob/330acf8a7c2ac6c484e26140d1fb2ffa19098935/openrr-gui/src/style.rs#L6) here.

```suggestion
            style::theme(None).unwrap(),
```